internal/dataplane/sendconfig/config_getter.go

```diff
@@ -85,6 +91,14 @@ func (d *DefaultConfigurationChangeDetector) HasConfigurationChanged(
 	return false, nil
 }
 
+func (d *DefaultConfigurationGetter) GetCurrentStatus(ctx context.Context, statusClient StatusClient) (*kong.Status, error) {
+	status, err := statusClient.Status(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return status, nil
+}
```

Comment:
Why do we need this method? It's just passing the `statusClient.Status` result. Maybe we could get it done without modifying `ConfigurationChangeDetector` and just use `go-kong.Client.Status` up in `KongClient` (we can use the `StatusClient` interface there to make it abstract).